### PR TITLE
Usages tooltip scope

### LINKED ISSUE #4200: Usages tooltip doesn't show function name
## Problem description
When I Cmd+Click on struct field, I get small window with usages of that field
изображение

But there are no function and struct names, like in PyCharm
изображение

### PR DESCRIPTION
**Pull request by fan-tom**
Closes #4200.

Adds function/method name to usages tooltip.

изображение

### CODE REVIEW

File: src/main/kotlin/org/rust/ide/search/RsFunctionGroupingRuleProvider.kt [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import com.intellij.usages.rules.UsageGroupingRule
+import org.rust.lang.core.psi.RsFunction
+import org.rust.lang.core.psi.RsImplItem
+
+class RsFunctionGroupingRuleProvider : FileStructureGroupRuleProvider {
+    override fun getUsageGroupingRule(project: Project): UsageGroupingRule? {
+        return RsFunctionGroupingRule()
+    }
+
+    private class RsFunctionGroupingRule : SingleParentUsageGroupingRule() {
```
Comment (mchernyavsky) [inline src/main/kotlin/org/rust/ide/search/RsFunctionGroupingRuleProvider.kt]: Make it staitc (move to `companion object { ... }`).

File: src/main/kotlin/org/rust/ide/search/RsFunctionGroupingRuleProvider.kt [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        override fun getParentGroupFor(usage: Usage, targets: Array<UsageTarget>): UsageGroup? {
+            if (usage !is PsiElementUsage) return null
+            val psiElement = usage.element
+            val rsFunction = PsiTreeUtil.getParentOfType(
```
Comment (mchernyavsky) [inline src/main/kotlin/org/rust/ide/search/RsFunctionGroupingRuleProvider.kt]: You can use `ancestorOrSelf`:
```kotlin
override fun getParentGroupFor(usage: Usage, targets: Array<UsageTarget>): UsageGroup? {
    if (usage !is PsiElementUsage) return null
    val rsFunction = usage.element.ancestorOrSelf<RsFunction>(RsImplItem::class.java) ?: return null
    return PsiNamedElementUsageGroupBase(rsFunction)
}
```

File: src/main/kotlin/org/rust/ide/search/RsFunctionGroupingRuleProvider.kt (hunk 1 above)
Comment (Undin) [reply src/main/kotlin/org/rust/ide/search/RsFunctionGroupingRuleProvider.kt]: @mchernyavsky Are you about `RsFunctionGroupingRule` class?
If yes then it's already static

**Comment by Undin** [comment]
bors retry

**Pull request merged**